apps/projects/app/components/Panel/NewProject/RepoSelector.js

```diff
@@ -128,7 +132,11 @@ class Repo extends React.Component {
         </div>
       )
 
-    const repositories = repos.length > 0 ? repoList : <Loading />
+    // if there are repos to add - list them. otherwise either all repos
+    // are already added or we're still waiting for the list from external
+    const repositories = repos.length > 0 ? repoList :
```

Comment:
Can we maybe just make this into a function? The chained ternary is sort of hard to read. I'm fine keeping it as is if you think it's better this way but to me it seems like a little too much logic to chain.

Reply:
You are right, nested ternaries are nasty. Improved (and saved myself from a lot of future confusion).